### ISSUE TITLE
cheatcode - Strict timing constraints can cause proposals to get stuck due to missed execution windows

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2024-01-olympus-on-chain-governance-judging#105]
cheatcode

medium

# Strict timing constraints can cause proposals to get stuck due to missed execution windows

If the execution window is missed (e.g if execute is not called in time after eta), the proposal might be stuck in a limbo state, neither executable nor cancellable. This could be mitigated by including a grace period or a mechanism to re-queue proposals under certain conditions.

### Lines of Code

https://github.com/sherlock-audit/2024-01-olympus-on-chain-governance/blob/main/bophades/src/external/governance/GovernorBravoDelegate.sol#L255C5-L278C6

```solidity
function execute(uint256 proposalId) external payable {
    // Check if the proposal is in Queued state
    if (state(proposalId) != ProposalState.Queued) {
        revert GovernorBravo_Execute_NotQueued();
    }

    Proposal storage proposal = proposals[proposalId];
    // Ensure the proposal is executed only after the time lock delay
    if (block.timestamp < proposal.eta) {
        revert GovernorBravo_Execute_TimelockNotPassed();
    }

    // Execute proposal actions

}
```

### Mitigation
Introduce a Grace Period after the `eta` has passed. This allows for a window of time during which the proposal can still be executed, even if the exact `eta` is missed. 

```solidity
function execute(uint256 proposalId) external payable {
    if (state(proposalId) != ProposalState.Queued) {
        revert GovernorBravo_Execute_NotQueued();
    }

    Proposal storage proposal = proposals[proposalId];
    uint256 gracePeriod = 48 hours;  // Example grace period

    // Check if current time is within the grace period after eta
    if (block.timestamp < proposal.eta || block.timestamp > proposal.eta + gracePeriod) {
        revert GovernorBravo_Execute_WindowMissed();
    }

    // Execute proposal actions
    
}
```

The grace period should be carefully chosen to prevent misuse (e.g delaying execution for strategic reasons). It should be long enough to allow execution but short enough to maintain the timeliness of governance actions.

Duplicate of #114
